cmd/cluster/agent/create.go

```diff
@@ -61,11 +61,10 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 		infraID = fmt.Sprintf("%s-%s", opts.Name, utilrand.String(5))
 	}
 	exampleOptions.InfraID = infraID
-
-	// TODO: We are inconsistent atm as DNS basedomain input is required and platform agnostic at the HostedCluster API level.
-	// However In the CLI input is only exposed in AWS and not required atm. This should either be core input and required for
-	// every platform or the field in the API should be optional if there's platform which don't need it.
-	exampleOptions.BaseDomain = "example.com"
+	exampleOptions.BaseDomain = opts.BaseDomain
+	if exampleOptions.BaseDomain == "" {
```

Comment:
how does this work with "example.com"? is it because it's basically not used?

Reply:
Yes, same as none platform - https://github.com/openshift/hypershift/commit/a3b7812c2841d411be3d9458d10e076a7f4e1245

Reply:
yeh, it feels weird, but I guess this is ok as a consequence of the services exposure strategy, at least until we develop more feedback